Cubes (#19)
compprog-lecture-tools/fun-contest-2023: cp23/hpi-fun2/cubes/problem.tex

\input{template.tex}

\begin{document}

\makeheader

``Setz dich nicht so nah vor den Fernseher - du bekommst noch viereckige Augen!''

> ``Don't sit so close to the TV - you'll get square eyes!''

Reviewer: (You may want to take a look at the `epigraph` package)
![Screenshot from 2023-07-08 14-04-16](https://github.com/compprog-lecture-tools/fun-contest-2023/assets/33495148/6c2dfceb-0c9f-444f-a242-4280f267ddc4)

You were told often enough by your parents, but of course you didn't believe them.

> Now, staring on computer screen all day to solve compettive programming tasks, the threat has turned into bitter reality.

Reviewer: ```suggestion
Now, staring on computer screens all day to solve competitive programming tasks, the threat has turned into bitter reality.
```

> You only see the world as 1cm x 1cm x 1cm large aligned blocks, even after switching profession to be a meterologist, your sight hasn't cleared.

Reviewer: ```suggestion
You only see the world as 1cm x 1cm x 1cm large aligned blocks, even after switching profession to be a meteorologist, your sight hasn't cleared.
```

> One day, your new boss asks you to calculate the surface area of a large cloud on the sky, so the National Weather Modification Service can shoot the right amount of Silver iodide into the sky to let the cloud rain down.

Reviewer: ```suggestion
One day, your new boss asks you to calculate the surface area of a large cloud in the sky, so the National Weather Modification Service can shoot the right amount of silver iodide into the sky to let the cloud rain down.
```

> Can you calculate the surface area of the water drops in the cloud, which you only see as little cubes?

Reviewer: But isn't it more the surface area of the cloud (consisting of the little cubes)?

Also more important:
Is the cloud connected?

\paragraph*{Input}

> In the first line, you get $n \leq 10^5$ - number of water drop blocks.

Reviewer: Please add a lower bound.

> In every one in the next $n$ lines you get a list of integer coordinates $(x, y, z)$ of each of the water blocks. The coordinates are non-negatvive integers smaller than $10^{12}$.

Reviewer: ```suggestion
In every one of the next $n$ lines you get a list of integer coordinates $(x, y, z)$ of each of the water blocks. The coordinates are non-negative integers smaller than $10^{12}$.
```
or something like
```suggestion
Each of the next $n$ lines contain 3 integer coordinates $1 \leq x, y, z \leq 10^{12}$ of a water block.
```

\paragraph*{Output}

> The surface area of the water drops in the cloud.

Reviewer: It is the surface area of the cloud, right?

\input{template.tex}

\usepackage{epigraph}

\begin{document}

\makeheader

\epigraph{``Setz dich nicht so nah vor den Fernseher - du bekommst noch
  viereckige Augen!\\ \textit{Don't sit so close to the TV - you'll get square
    eyes!}''}{Your parents}

You were told often enough by your parents, but of course you didn't believe them.

Now, staring on computer screens all day to solve competitive programming tasks, the threat has turned into bitter reality.

You only see the world as 1cm x 1cm x 1cm large aligned blocks, even after switching profession to be a meteorologist, your sight hasn't cleared.

One day, your new boss asks you to calculate the surface area of a large cloud in the sky, so the National Weather Modification Service can shoot the right amount of silver iodide into the sky to let the cloud rain down.

> It might be no cloud at all if the weather is good, but if it is a cloud, it can be also disconnected.

Reviewer: ```suggestion
There might be no cloud at all if the weather is good, but if there is a cloud, it can also be disconnected.
```

> Can you calculate the surface area of the water drops in the cloud, which you only see as little cubes?

Reviewer: ```suggestion
Can you calculate the surface area of the cloud, which you only see as little cubes?
```

\input{template.tex}

\usepackage{epigraph}

\begin{document}

\makeheader

\epigraph{``Setz dich nicht so nah vor den Fernseher - du bekommst noch
  viereckige Augen!\\ \textit{Don't sit so close to the TV - you'll get square
    eyes!}''}{Your parents}

You were told often enough by your parents, but of course you didn't believe them.

Now, staring on computer screens all day to solve competitive programming tasks, the threat has turned into bitter reality.

You only see the world as 1cm x 1cm x 1cm large aligned blocks, even after switching profession to be a meteorologist, your sight hasn't cleared.

One day, your new boss asks you to calculate the surface area of a large cloud in the sky, so the National Weather Modification Service can shoot the right amount of silver iodide into the sky to let the cloud rain down.

It might be no cloud at all if the weather is good, but if it is a cloud, it can be also disconnected.

Can you calculate the surface area of the water drops in the cloud, which you only see as little cubes?

\paragraph*{Input}

In the first line, you get $0 \leq n \leq 10^5$ - number of water drop blocks.

> Each of the next $n$ lines contain 3 integer coordinates $0 \leq x, y, z \leq 10^{12}$ of a water block. It is not garanteed that the blocks are connected, but they are pairwise distinct.

Reviewer: ```suggestion
Each of the next $n$ lines contains 3 integer coordinates $0 \leq x, y, z \leq 10^{12}$ of a water block. It is not guaranteed that the blocks are connected, but they are pairwise distinct.
```

The first one is my fault ^^'
